proto-build/src/code_generator.rs

```diff
@@ -0,0 +1,280 @@
+            self.project.name
+        );
+
+        self.transform();
+        self.generate_mod_file();
+        self.fmt();
+
+        info!(
+            "✨  [{}] Library is successfully generated!",
+            self.project.name
+        );
+    }
+
+    fn prepare_dir(&self) {
+        if self.tmp_build_dir.exists() {
+            remove_dir_all(self.tmp_build_dir.clone()).unwrap();
+        }
+        create_dir_all(self.tmp_namespaced_dir()).unwrap();
+        output_version_file(
+            &self.project.name,
+            &self.project.version,
+            &self.tmp_namespaced_dir(),
+        );
+    }
+
+    fn generate_mod_file(&self) {
+        mod_gen::generate_mod_file(&self.absolute_out_dir());
+    }
+
+    fn transform(&self) {
+        transform::copy_and_transform_all(
+            &self.tmp_namespaced_dir(),
+            &self.absolute_out_dir(),
+            &self.file_descriptor_set(),
+        );
+    }
+
+    fn absolute_out_dir(&self) -> PathBuf {
+        self.root.join(&self.out_dir)
+    }
+
+    fn fmt(&self) {
+        let manifest_path = find_cargo_toml(&self.absolute_out_dir());
+        let exit_status = Command::new("cargo")
+            .arg("fmt")
+            .arg("--manifest-path")
+            .arg(manifest_path.to_string_lossy().to_string())
+            .spawn()
+            .unwrap()
+            .wait()
+            .unwrap();
+
+        if !exit_status.success() {
+            panic!(
+                "unable to format with: cargo fmt --manifest-path {}",
+                manifest_path.to_string_lossy()
+            );
+        }
+    }
+
+    fn compile_proto(&self) {
+        let mut all_related_projects = self.deps.clone();
+        all_related_projects.push(self.project.clone());
+
+        let buf_gen_template = self.root.join("buf.gen.yaml");
+
+        info!(
+            "🧪 [{}] Compiling types from protobuf definitions...",
+            self.project.name
+        );
+
+        // Compile proto files for each file in `protos` variable
+        // `buf generate —template {<buf_gen_template} <proto_file>`
+        for project in all_related_projects {
+            let buf_root = if project.name == "cosmos" || project.name == "ics23" {
+                self.root.join(&project.project_dir).join("proto")
+            } else {
+                WalkDir::new(&self.root.join(&project.project_dir))
+                    .into_iter()
+                    .filter_map(|e| e.ok())
+                    .find(|e| {
+                        e.file_name()
+                            .to_str()
+                            .map(|s| s == "buf.yaml" || s == "buf.yml")
+                            .unwrap_or(false)
+                    })
+                    .map(|e| e.path().parent().unwrap().to_path_buf())
+                    .unwrap()
+            };
+            // let buf_root = WalkDir::new(&self.root.join(&project.project_dir))
```

Comment:
Commented code, remove it please, if it's not needed

Reply:
Removed